### PR TITLE
Mahara PR #2

### PR DESCRIPTION
**Pull request by darrin2016**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Removed references to all non-Mahara code and references
* Update stack image and documentation
* Some URLs in the README.md will be incorrect until merged:
  * Deploy to Azure link
  * Visualize link
  * Issues link
  * Repo link

The point '"Has anyone run this template sucessfully in production? Yes they have. With that being said we do not make any performance guarantees about this architecture"' has been left in the README.md, should this be removed as part of the PR ?

The content URL for github has been left as the MS Azure links since the previous PR review suggested to update it as such. See https://github.com/Azure/azure-quickstart-templates/pull/4283#discussion_r177544596 

Does Mahara come under the banner of a 'managedApplication' or should that reference (and directory) be removed as part of the PR ?

Thanks

Darrin

### CODE REVIEW

**Comment by rgardler** [comment]
@bmoore-msft Note I'm working with the provider of this PR via email.

**Comment by darrin2016** [comment]
CI is putting out an error that I can't work out what its referring too:

>   1 failing
>  1) Template Local Validation mahara-autoscale-cache metadata.json matches schema:
>     AssertionError: #:Missing required property "$schema"

As far as I can see, the metadata.json appears correct, so I'm not sure where the error is, anybody have any ideas?

**Comment by darrin2016** [comment]
The JSON files seem to pass validation (at least by a generic JSON validator). Have not been able to find out what the CI error means or relates to.

> for i in `find . -name '*.json'`;do echo "Checking $i";jsonlint-php $i;done
> Checking ./azuredeploy.json
> Valid JSON
> Checking ./metadata.json
> Valid JSON
> Checking ./nested/elasticconfig.json
> Valid JSON
> Checking ./nested/network.json
> Valid JSON
> Checking ./nested/postgres.json
> Valid JSON
> Checking ./nested/mysql.json
> Valid JSON
> Checking ./nested/recoveryservices.json
> Valid JSON
> Checking ./nested/glustervm.json
> Valid JSON
> Checking ./nested/controller.json
> Valid JSON
> Checking ./nested/elastic-search.json
> Valid JSON
> Checking ./nested/none-search.json
> Valid JSON
> Checking ./nested/recoveryservicesEnlist.json
> Valid JSON
> Checking ./nested/storageAccount.json
> Valid JSON
> Checking ./nested/controllerconfig.json
> Valid JSON
> Checking ./nested/glustervmconfig.json
> Valid JSON
> Checking ./nested/webvmss.json
> Valid JSON
> Checking ./nested/gluster.json
> Valid JSON
> Checking ./azuredeploy.parameters.json
> Valid JSON

File: mahara-autoscale-cache/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# *Deploy and manage a Scalable Mahara Cluster on Azure*
+
+After deploying, these templates will provide you with a new Mahara site with caching for speed and scaling frontends to handle PHP load. The filesystem behind it is mirrored for high availability. Filesystem permissions and options have been tuned to make Mahara more secure than the default install
```
[the remark this answers is not in the document]
Comment (darrin2016) [reply mahara-autoscale-cache/README.md]: All of the text changes have been made.

File: mahara-autoscale-cache/azuredeploy.json [hunk 2]
```diff
@@ -360,7 +365,7 @@
     "resources": [
         {
             "type": "Microsoft.Resources/deployments",
-            "apiVersion": "2017-05-10",
+            "apiVersion": "2016-09-01",
             "name": "pid-738e3eec-68d4-4667-8377-c05c77c21f1b",
```
Comment (bmoore-msft) [inline mahara-autoscale-cache/azuredeploy.json]: @rgardler - is this correct?

File: mahara-autoscale-cache/scripts/helper_functions.sh [hunk 3]
```diff
@@ -0,0 +1,753 @@
...
+
+    cat <<EOF > /etc/mahara_azure_files.credential
+username=$storageAccountName
+password=$storageAccountKey
+EOF
+    chmod 600 /etc/mahara_azure_files.credential
+    
+    grep "^//$storageAccountName.file.core.windows.net/mahara\s\s*/mahara\s\s*cifs" /etc/fstab
+    if [ $? != "0" ]; then
+        echo -e "\n//$storageAccountName.file.core.windows.net/mahara   /mahara cifs    credentials=/etc/mahara_azure_files.credential,uid=www-data,gid=www-data,nofail,vers=3.0,dir_mode=0770,file_mode=0660,serverino,mfsymlinks" >> /etc/fstab
```
[the remark this answers is not in the document]
Comment (bmoore-msft) [reply mahara-autoscale-cache/scripts/helper_functions.sh]: Still needs to be addressed

File: mahara-autoscale-cache/scripts/helper_functions.sh [hunk 4]
```diff
@@ -0,0 +1,753 @@
...
+    local storageAccountName=$1
+    local storageAccountKey=$2
+
+    cat <<EOF > /etc/mahara_azure_files.credential
+username=$storageAccountName
+password=$storageAccountKey
+EOF
+    chmod 600 /etc/mahara_azure_files.credential
+    
+    grep "^//$storageAccountName.file.core.windows.net/mahara\s\s*/mahara\s\s*cifs" /etc/fstab
```
[the remark this answers is not in the document]
Comment (bmoore-msft) [reply mahara-autoscale-cache/scripts/helper_functions.sh]: still needs to be addressed

File: mahara-autoscale-cache/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,734 @@
...
+                "description": "Virtual network gateway vpn type"
+            },
+            "type": "string"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2016-09-01",
+            "name": "pid-738e3eec-68d4-4667-8377-c05c77c21f1b",
```
Comment (bmoore-msft) [inline mahara-autoscale-cache/azuredeploy.json]: @rgardler - is this correct?

File: mahara-autoscale-cache/scripts/helper_functions.sh [hunk 6]
```diff
@@ -0,0 +1,765 @@
...
+{
+    local storageAccountName=$1
+    local storageAccountKey=$2
+
+    cat <<EOF > /etc/mahara_azure_files.credential
+username=$storageAccountName
+password=$storageAccountKey
+EOF
+    chmod 600 /etc/mahara_azure_files.credential
+
```
Comment (bmoore-msft) [inline mahara-autoscale-cache/scripts/helper_functions.sh]: still need these hardcoded endpoints fixed...

**Comment by bmoore-msft** [comment]
@rgardler - one comment in here for you...

File: mahara-autoscale-cache/azuredeploy.json (hunk 5 above)
Comment (rgardler) [reply mahara-autoscale-cache/azuredeploy.json]: No, this is not correct. This has been copied from a different template and should be removed from here. Thanks.

File: mahara-autoscale-cache/scripts/helper_functions.sh (hunk 6 above)
Comment (darrin2016) [reply mahara-autoscale-cache/scripts/helper_functions.sh]: Yes, I raised an issue about this, please see https://github.com/Azure/azure-quickstart-templates/issues/4960

**Pull request merged**